### PR TITLE
Avoid panic with improperly encoded secrets

### PR DESCRIPTION
**Pull request by fitzthum**
Fixes: #55

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

### CODE REVIEW

**Comment by fitzthum** [comment]
btw @dubek @wainersm

File: src/request.rs [hunk 1]
```diff
@@ -162,8 +162,12 @@ pub struct Key {
 }
 
 impl Key {
-    pub fn into_bytes(&self) -> Vec<u8> {
-        base64::decode(&self.payload).unwrap()
+    pub fn bytes(&self) -> Result<Vec<u8>> {
+        base64::decode(&self.payload).map_err(|e| {
+            // Provide full details in KBS log
+            error!("Binary secrets must be b64 encoded: {}", e);
```
Comment (dubek) [inline src/request.rs:168]: s/b64/base64/ 

(to be clearer)

File: src/request.rs [hunk 2]
```diff
@@ -122,7 +122,7 @@ impl SecretType for SecretKey {
     async fn payload(&self, db: &KbsDb, connection: Connection) -> Result<Vec<u8>> {
         let key = db.get_secret(&self.request.id).await?;
         Ok(match &self.request.format[..] {
-            "binary" => key.into_bytes(),
+            "binary" => key.bytes()?,
             "json" => serde_json::to_string(&key).unwrap().into_bytes(),
```
Comment (dubek) [inline src/request.rs:126]: 1. we should tackle at this `unwrap()` as well -- if there's invalid JSON string in the database, this `unwrap()` will panic the entire server.
2. we should add a test which populates invalid values into the DB and check that error is returned.

(both can wait for another PR)

**Review by dubek — APPROVED** [review]
Looks good.
Couple of small comments:

File: src/db.rs [hunk 3]
```diff
@@ -98,7 +92,7 @@ impl KbsDb {
         let mut counter = 0;
         let result = question_mark_re.replace_all(sql, |_: &Captures| {
             counter += 1;
-            format!("${}", counter)
+            format!("${counter}")
```
Comment (wainersm) [inline src/db.rs:94]: is this `$` sign needed?

Comment (dubek) [reply src/db.rs:94]: Postgres variable bindings (simple `?` in mysql land) are `$1`, `$2`, `$3` , etc.

Comment (wainersm) [reply src/db.rs:94]: ah, ok. thanks @dubek !

**Review by wainersm — APPROVED** [review]
I don't have any other suggestions other than of @dubek 's

File: src/request.rs (hunk 2 above)
Comment (fitzthum) [reply src/request.rs:126]: Yeah, I will do a follow-up addressing some other unwraps and a few other cleanup things I noticed .

**Pull request merged**